### PR TITLE
Initial google ml-engine support

### LINKED ISSUE #69: Add ML Engine model
Add ML engine model to abstract away serving from ML Engine. Users would push models to ML Engine and still use Zoltar to get predictions, zoltar would make call to ML Engine and retrieve prediction.

### PR DESCRIPTION
**Pull request by regadas**
Fixes #69 

@ravwojdyla PTAL

### CODE REVIEW

**Comment by regadas** [comment]
@ravwojdyla PTAL

File: zoltar-mlengine/src/main/java/com/spotify/zoltar/mlengine/MlEngineModel.java [hunk 1]
```diff
@@ -0,0 +1,112 @@
...
+  abstract GoogleCredential credentials();
+
+  abstract JsonFactory jsonFactory();
+
+  /**
+   * Predict.
+   *
+   * @param data prediction input data.
+   */
+  public <T> List<T> predict(final List<?> data) throws IOException {
```
Comment (ravwojdyla) [inline zoltar-mlengine/src/main/java/com/spotify/zoltar/mlengine/MlEngineModel.java]: Should we expose `predict` as `List<T>` or `List<HashMap<?, ?>>` - is it always a list of a map in response json? Or maybe we could also have a method that takes a class for json parser of response? Would that work?

File: zoltar-mlengine/src/main/java/com/spotify/zoltar/mlengine/MlEngineModel.java [hunk 2]
```diff
@@ -0,0 +1,112 @@
...
+   *           <pre>"projects/{PROJECT_ID}/models/{MODEL_ID}"</pre>
+   *           or
+   *           <pre>"projects/{PROJECT_ID}/models/{MODEL_ID}/versions/{MODEL_VERSION}"</pre>
+   */
+  public static MlEngineModel create(final Model.Id id) throws Exception {
+    final HttpTransport httpTransport = GoogleNetHttpTransport.newTrustedTransport();
+    final JsonFactory jsonFactory = JacksonFactory.getDefaultInstance();
+    final GoogleCredential credential = GoogleCredential.getApplicationDefault();
+    final CloudMachineLearningEngine mlEngine = new CloudMachineLearningEngine
+        .Builder(httpTransport, jsonFactory, null)
```
[the remark this answers is not in the document]
Comment (regadas) [reply zoltar-mlengine/src/main/java/com/spotify/zoltar/mlengine/MlEngineModel.java]: @ravwojdyla I think it's ok to be explicit here.

File: zoltar-mlengine/src/main/java/com/spotify/zoltar/mlengine/MlEngineModel.java [hunk 3]
```diff
@@ -0,0 +1,110 @@
...
+   *
+   * @param id {@link Model.Id} needs to be created with the following format:
+   *           <pre>"projects/{PROJECT_ID}/models/{MODEL_ID}/versions/{MODEL_VERSION}"</pre>
+   */
+  public static MlEngineModel create(final Model.Id id) throws Exception {
+    final HttpTransport httpTransport = GoogleNetHttpTransport.newTrustedTransport();
+    final JsonFactory jsonFactory = JacksonFactory.getDefaultInstance();
+    final GoogleCredential credential = GoogleCredential.getApplicationDefault();
+    final CloudMachineLearningEngine mlEngine = new CloudMachineLearningEngine
+        .Builder(httpTransport, jsonFactory, null)
```
[the remark this answers is not in the document]
Comment (regadas) [reply zoltar-mlengine/src/main/java/com/spotify/zoltar/mlengine/MlEngineModel.java]: @ravwojdyla good I missed that!

File: zoltar-mlengine/src/main/java/com/spotify/zoltar/mlengine/MlEngineModel.java [hunk 4]
```diff
@@ -0,0 +1,110 @@
...
+public abstract class MlEngineModel implements Model<CloudMachineLearningEngine> {
+
+  /**
+   * Creates a Google Cloud ML Engine backed model.
+   *
+   * @param id {@link Model.Id} needs to be created with the following format:
+   *           <pre>"projects/{PROJECT_ID}/models/{MODEL_ID}/versions/{MODEL_VERSION}"</pre>
+   */
+  public static MlEngineModel create(final Model.Id id) throws Exception {
+    final HttpTransport httpTransport = GoogleNetHttpTransport.newTrustedTransport();
```
[the remark this answers is not in the document]
Comment (regadas) [reply zoltar-mlengine/src/main/java/com/spotify/zoltar/mlengine/MlEngineModel.java]: @ravwojdyla well the default one does not include the certificates as this one.

File: zoltar-mlengine/src/main/java/com/spotify/zoltar/mlengine/MlEngineModel.java [hunk 5]
```diff
@@ -0,0 +1,110 @@
...
+    final JsonFactory jsonFactory = JacksonFactory.getDefaultInstance();
```
[the remark this answers is not in the document]
Comment (regadas) [reply zoltar-mlengine/src/main/java/com/spotify/zoltar/mlengine/MlEngineModel.java]: @ravwojdyla this one I think it's ok

File: zoltar-mlengine/src/main/java/com/spotify/zoltar/mlengine/MlEngineModel.java [hunk 6]
```diff
@@ -0,0 +1,188 @@
...
+
+      /**
+       * List of predictions.
+       *
+       * @param klass class to each returned prediction objects are converted.
+       */
+      public <T> List<T> values(final Class<T> klass) {
+        return values()
+            .stream()
+            .map(p -> MAPPER.convertValue(p, klass))
```
Comment (ravwojdyla) [inline zoltar-mlengine/src/main/java/com/spotify/zoltar/mlengine/MlEngineModel.java]: Is some kind of memorization of `MAPPER` for specific `klass` needed here?

Comment (regadas) [reply zoltar-mlengine/src/main/java/com/spotify/zoltar/mlengine/MlEngineModel.java]: good point, but i'm not sure if can do it, let me have a look.

**Pull request merged**